src/main/java/org/zaproxy/zap/extension/jwt/attacks/SignatureAttack.java

```diff
@@ -143,6 +143,31 @@ private boolean executePubliclyWellKnownHMacSecretAttack() {
         return false;
     }
 
+
+
+    private boolean executeIncorrectSignatureAttack() {
+        JWTHolder cloneJWTHolder = new JWTHolder(this.serverSideAttack.getJwtHolder());
+        JSONObject payloadJSONObject = new JSONObject(cloneJWTHolder.getPayload());
+        payloadJSONObject.put("username", "admin");
+        clonedJWTHolder.setPayload(payloadJSONObject.toString());
+
+        if (this.serverSideAttack.getJwtActiveScanRule().isStop()) {
+            return false;
+        }
+
+        if (verifyJWTToken(cloneJWTHolder.getBase64EncodedToken(), serverSideAttack)) {
+            raiseAlert(
+                    MESSAGE_PREFIX,
+                    VulnerabilityType.NULL_BYTE,
```

Comment:
Please create a new [VulnerabilityType](https://github.com/SasanLabs/owasp-zap-jwt-addon/blob/cb0b85efa9c291e3a5eeb35b755e7fbc35c4f2bc/src/main/java/org/zaproxy/zap/extension/jwt/utils/VulnerabilityType.java#L20) and use it here.

Reply:
Please also build the message to show in the alert like [Message](https://github.com/SasanLabs/owasp-zap-jwt-addon/blob/595e00ad997688c494bdfeb7819dd102619417f1/src/main/resources/org/zaproxy/zap/extension/jwt/resources/Messages.properties#L72)

Reply:
Updated @preetkaran20